lib/commands/media-projection.js

```diff
@@ -0,0 +1,270 @@
+        resolve();
+      }, RECORDING_STARTUP_TIMEOUT_MS);
+    });
+    return true;
+  }
+
+  async cleanup () {
+    await this.adb.shell([`rm -f ${RECORDINGS_ROOT}/*`]);
+  }
+
+  async pullRecent () {
+    const recordings = await this.adb.ls(RECORDINGS_ROOT, ['-tr']);
+    if (_.isEmpty(recordings)) {
+      return null;
+    }
+
+    const dstPath = path.join(await tempDir.openDir(), recordings[0]);
+    await this.adb.pull(`${RECORDINGS_ROOT}/${recordings[0]}`, dstPath);
+    return dstPath;
+  }
+
+  async stop () {
+    if (!await this.isRunning()) {
+      return false;
+    }
+
+    await this.adb.shell([
+      'am', 'start',
+      '-n', RECORDING_ACTIVITY_NAME,
+      '-a', RECORDING_ACTION_STOP,
+    ]);
+    try {
+      await waitForCondition(async () => !(await this.isRunning()), {
+        waitMs: RECORDING_STOP_TIMEOUT_MS,
+        intervalMs: 500,
+      });
+    } catch (e) {
+      throw new Error(
+        `The current media projection recording cannot stop within ${RECORDING_STOP_TIMEOUT_MS}ms timeout`
+      );
+    }
+    return true;
+  }
+}
+
+
+/**
+ * @typedef {Object} StartRecordingOptions
+ *
+ * @property {string?} resolution Maximum supported resolution on-device (Detected
+ * automatically by the app itself), which usually equals to Full HD 1920x1080 on most
+ * phones however you can change it to following supported resolutions
+ * as well: "1920x1080", "1280x720", "720x480", "320x240", "176x144".
+ * @property {number?} maxDurationSec [900] Default value: 900 seconds which means
+ * maximum allowed duration is 15 minute, you can increase it if your test takes
+ * longer than that.
+ * @property {string?} priority [high] Means recording thread priority is maximum
+ * however if you face performance drops during testing with recording enabled, you
+ * can reduce recording priority to "normal" or "low".
+ * @property {string?} filename You can type recording video file name as you want,
+ * but recording currently supports only "mp4" format so your filename must end with ".mp4".
+ * An invalid file name will fail to start the recording.
+ */
+
+/**
+ * Record the display of a real devices running Android 10 (API level 29) and higher.
+ * The screen activity is recorded to a MPEG-4 file. Audio is also recorded by default
+ * (only for apps that allow it in their manifests).
+ * If another recording has been already started then the command will exit silently.
+ * The previously recorded video file is deleted when a new recording session is started.
+ * Recording continues on driver session deletion if not stopped explicitly until the
+ * timeout happens.
+ *
+ * @param {?StartRecordingOptions} options Available options.
+ * @returns {boolean} True if a new recording has successfully started.
+ * @throws {Error} If recording has failed to start or is not supported on the device under test.
+ */
+commands.mobileStartMediaProjectionRecording = async function mobileStartMediaProjectionRecording (options = {}) {
```

Comment:
"mediaProjectionRecording" is an awful long string to have inside names everywhere :-(

Reply:
I could use mpRecording, not sure if it would be readable enough though

Reply:
maybe for the internal code `mp` is fine, and keep the "mediaProjection" for the user-facing strings? not sure if that's easy to do.

it's not a huge deal, i just hate reading entire sentences as variable names :-)